### PR TITLE
XDR-1701: add azurerm_sentinel_watchlist & azurerm_sentinel_watchlist_item modules

### PR DESCRIPTION
**Pull request by chrisjaimon2012**

### CODE REVIEW

**Comment by chrisjaimon2012** [comment]
<details><summary>Terraform apply output for <code>azurerm_sentinel_watchlist</code></summary><p>

```robots.txt
$ terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_sentinel_watchlist.watchlist will be created
  + resource "azurerm_sentinel_watchlist" "watchlist" {
      + display_name               = "test"
      + id                         = (known after apply)
      + item_search_key            = "Key"
      + log_analytics_workspace_id = "/subscriptions/00000000-0000-0000-0000-00000000000/resourceGroups/test-rg/providers/Microsoft.OperationalInsights/workspaces/test"
      + name                       = "test"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

azurerm_sentinel_watchlist.watchlist: Creating...
azurerm_sentinel_watchlist.watchlist: Still creating... [10s elapsed]
azurerm_sentinel_watchlist.watchlist: Creation complete after 10s [id=/subscriptions/00000000-0000-0000-0000-00000000000/resourceGroups/test-rg/providers/Microsoft.OperationalInsights/workspaces/test/providers/Microsoft.SecurityInsights/watchlists/test]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

id = /subscriptions/00000000-0000-0000-0000-00000000000/resourceGroups/test-rg/providers/Microsoft.OperationalInsights/workspaces/test/providers/Microsoft.SecurityInsights/watchlists/test
```
</p></details>

<details><summary>Terraform apply output for <code>azurerm_sentinel_watchlist_item</code></summary><p>

```robots.txt
$ terraform apply

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # azurerm_sentinel_watchlist_item.watchlist_item will be created
  + resource "azurerm_sentinel_watchlist_item" "watchlist_item" {
      + id           = (known after apply)
      + name         = (known after apply)
      + properties   = {
          + "Key" = "v1"
          + "k2"  = "v2"
        }
      + watchlist_id = "/subscriptions/00000000-0000-0000-0000-00000000000/resourceGroups/test-rg/providers/Microsoft.OperationalInsights/workspaces/test/providers/Microsoft.SecurityInsights/watchlists/test"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + id = (known after apply)

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

azurerm_sentinel_watchlist_item.watchlist_item: Creating...
azurerm_sentinel_watchlist_item.watchlist_item: Creation complete after 7s [id=/subscriptions/00000000-0000-0000-0000-00000000000/resourceGroups/test-rg/providers/Microsoft.OperationalInsights/workspaces/test/providers/Microsoft.SecurityInsights/watchlists/test/watchlistItems/8a00d0f0-977b-474f-a5fe-9f2413026129]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

id = /subscriptions/00000000-0000-0000-0000-00000000000/resourceGroups/test-rg/providers/Microsoft.OperationalInsights/workspaces/test/providers/Microsoft.SecurityInsights/watchlists/test/watchlistItems/8a00d0f0-977b-474f-a5fe-9f2413026129
```
</p></details>

File: modules/sentinel-watchlist-item/vars.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+variable "name" {
+  description = "The name in UUID format which should be used for this Sentinel Watchlist Item."
+  type        = string
+  default     = null
```
Comment (GerardSetho) [inline modules/sentinel-watchlist-item/vars.tf:4]: Should this default to null? 

Shouldn't this be a required var?

Comment (chrisjaimon2012) [reply modules/sentinel-watchlist-item/vars.tf:4]: Its null as default, just to maintain consistency with the terraform provider, where name is an optional value
https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/sentinel_watchlist_item#name

**Review by GerardSetho — APPROVED** [review]
Looks good!

**Pull request merged**